ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -227,10 +227,34 @@ - (void)addHandleToChain:(CKComponentScopeHandle *)handle
   _handles.push_back(handle);
 }
 
-- (id)responder
+- (CKScopedResponderKey)keyForHandle:(CKComponentScopeHandle *)handle
 {
+  static CKScopedResponderKey notFoundKey = INT_MAX;
+
+  if (handle == nil) {
+    return notFoundKey;
+  }
+
+  std::lock_guard<std::mutex> l(_mutex);
+  auto result = std::find(_handles.begin(), _handles.end(), handle);
+  if (result == _handles.end()) {
+    CKFailAssert(@"This scope handle is not associated with this Responder.");
+    return notFoundKey;
+  }
+
+  return (int)std::distance(_handles.begin(), result);
+}
+
+- (id)responderForKey:(CKScopedResponderKey)key
+{
+  if (key < 0) {
+    CKFailAssert(@"In valid key %d for responder.", key);
+    return nil;
+  }
+
   std::lock_guard<std::mutex> l(_mutex);
-  for (const auto &handle: _handles) {
+  for (int i=key; i<_handles.size(); i++) {
```

Comment:
If `key` is `INT_MAX` then this method just returns nil, right? Just making sure.